doc/ROADMAP.rst.txt

```diff
@@ -0,0 +1,377 @@
+it seems that the data point model is better defined and easier to implement.
+We therefore propose to move to this data representation for 1.0, and to vet
+all interpolation code to ensure that boundary values, transformations, etc.
+are correctly computed.  Addressing this issue will close several issues,
+including #1323, #1903, #2045 and #2640.
+
+The morphology interface needs to be standardized:
+
+  - binary dilation/erosion/opening/closing take a "structure" argument,
+    whereas their grey equivalent take size (has to be a tuple, not a scalar),
+    footprint, or structure.
+  - a scalar should be acceptable for size, equivalent to providing that same
+    value for each axis.
+  - for binary dilation/erosion/opening/closing, the structuring element is
+    optional, whereas it's mandatory for grey.  Grey morphology operations
+    should get the same default.
+  - other filters should also take that default value where possible.
+
+
+odr
+---
+Rename the module to ``regression`` or ``fitting``, include
+``optimize.curve_fit``. This module will then provide a home for other fitting
+functionality - what exactly needs to be worked out in more detail, a
+discussion can be found at https://github.com/scipy/scipy/pull/448.
+
+
+optimize
+````````
+Overall this module is in reasonably good shape, however it is missing a few
+more good global optimizers as well as large-scale optimizers.  These should be
+added.  Other things that are needed:
+
+  - deprecate the ``fmin_*`` functions in the documentation, ``minimize`` is
+    preferred.
+  - clearly define what's out of scope for this module.
+
+
+signal
+``````
+*Convolution and correlation*: (Relevant functions are convolve, correlate,
+fftconvolve, convolve2d, correlate2d, and sepfir2d.) Eliminate the overlap with
+`ndimage` (and elsewhere).  From `numpy`, `scipy.signal` and `scipy.ndimage`
+(and anywhere else we find them), pick the "best of class" for 1-D, 2-D and n-d
+convolution and correlation, put the implementation somewhere, and use that
+consistently throughout scipy.
+
+*B-splines*: (Relevant functions are bspline, cubic, quadratic, gauss_spline,
+cspline1d, qspline1d, cspline2d, qspline2d, cspline1d_eval, and spline_filter.)
+Move the good stuff to `interpolate` (with appropriate API changes to match how
+things are done in `interpolate`), and eliminate any duplication.
+
+*Filter design*: merge `firwin` and `firwin2` so `firwin2` can be removed.
+
+*Continuous-Time Linear Systems*: remove `lsim2`, `impulse2`, `step2`.  Make
+`lsim`, `impulse` and `step` "just work" for any input system.  Improve
+performance of ltisys (less internal transformations between different
+representations).
+
+*Wavelets*: what's there now doesn't make much sense.  Continous wavelets
+only at the moment - decide whether to completely rewrite or remove them.
+Discrete wavelet transforms should be added, see the "new modules" section
```

Comment:
I'm actually considering removing this. PyWavelets is now in pretty good shape, so could be used for this. However, it seems to be doing well as a standalone project and is not difficult to install. So maybe we should just leave things the way they are, and consider pulling it into SciPy only if development stalls for whatever reason.